### PR TITLE
[Confidential Ledger] Migrate to test-proxy

### PR DESCRIPTION
**Pull request by lynshi**
# Description

Migrate tests for Confidential Ledger to the new test-proxy framework. New recordings have been added, and I also tested with `AZURE_SKIP_LIVE_RECORDING=true` to ensure live tests that don't go through the proxy work.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/common/testproxy/target_version.txt [hunk 1]
```diff
@@ -1 +1 @@
-1.0.0-dev.20220805.3
+1.0.0-dev.20220810.2
```
Comment (lynshi) [inline eng/common/testproxy/target_version.txt:1]: Requires a tag that includes https://github.com/Azure/azure-sdk-tools/commit/2279cc0fa00f1e5e4035236f4d6a6aee5dfe3902

File: tools/azure-sdk-tools/devtools_testutils/sanitizers.py [hunk 2]
```diff
@@ -487,6 +488,9 @@ def _get_recording_option_args(**kwargs: "Any") -> dict:
         if tls_certificate:
             transport["TLSValidationCert"] = tls_certificate
 
+        if tls_certificate_host:
+            transport["TSLValidationCertHost"] = tls_certificate_host
```
Comment (lynshi) [inline tools/azure-sdk-tools/devtools_testutils/sanitizers.py:492]: Looks like a typo but matches [`azure-sdk-tools`](https://github.com/Azure/azure-sdk-tools/blob/e72dc36392d5eee4605ba3711586a6460880537b/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/TransportCustomizations.cs#L28)

**Comment by scbedd** [comment]
I have no earthly idea why an eng/common change is showing up in this PR. Super frustrating.

**Comment by lynshi** [comment]
> I have no earthly idea why an eng/common change is showing up in this PR. Super frustrating.

Fixed this, sorry! (The change was file mode 100644 -> 100755 because I `chmod +x` to run `apply-dev-cert.sh` - GitHub diff shows no file content changes, and the mode change is inline with the filename)

**Comment by scbedd** [comment]
@lynshi Can you put it back? I've overwritten the file locally with the one from `main`, but it still shows that change. We do not allow eng/common changes from anything other than a specific workflow.

**Review by scbedd — APPROVED** [review]

**Pull request merged**